carta/util.py

```diff
@@ -129,3 +134,214 @@ def split_action_path(path):
     """
     parts = path.split('.')
     return '.'.join(parts[:-1]), parts[-1]
+
+
+class PixelString:
+    """Parses pixel values."""
+
+    UNITS = {"px", "pix", "pixel", "pixels"}
+    UNIT_REGEX = rf"^(\d+(?:\.\d+)?)\s*(?:{'|'.join(UNITS)})$"
+
+    @classmethod
+    def valid(cls, value):
+        """Whether the input string is a numeric value followed by a pixel unit.
+
+        Permitted pixel unit strings are stored in :obj:`carta.util.PixelString.UNITS`. Whitespace is permitted after the number and before the unit.
+
+        Parameters
+        ----------
+        value : string
+            The input string.
+
+        Returns
+        -------
+        boolean
+            Whether the input string is a pixel value.
+        """
+        m = re.match(cls.UNIT_REGEX, value, re.IGNORECASE)
+        return m is not None
+
+    @classmethod
+    def normalized(cls, value):
+        """Parse a string containing a numeric value followed by a pixel unit, and return the numeric part (as a string).
+
+        Permitted pixel unit strings are stored in :obj:`carta.util.PixelString.UNITS`. Whitespace is permitted after the number and before the unit.
+
+        Parameters
+        ----------
+        value : string
+            The string representation of the pixel value.
+
+        Returns
+        -------
+        string
+            The numeric portion of the pixel value.
+
+        Raises
+        ------
+        ValueError
+            If the input string is not in a recognized format.
+        """
+        m = re.match(cls.UNIT_REGEX, value, re.IGNORECASE)
+        if m is None:
+            raise ValueError(f"{repr(value)} is not in a recognized pixel format.")
+        return m.group(1)
+
+
+class AngularSizeString:
+    """Parses angular sizes."""
+
+    NORMALIZED_UNIT = {
```

Comment:
I suggest we also support 
1. milliarcsec (10^-3 arcsec for VLA/ALMA etc)
2. microarcsec (10^-6 arcsec for VLBI, EHT etc)

Reply:
This is doable; I'll need to do a bit of refactoring to normalize these to one of the units accepted by the frontend. Are there any other aliases of these unit names that you would like to support?

Reply:
I added these (together with the full words, and `mas` and `µas`). I also added `uas`, since I know that `u` has sometimes been used as an ASCII approximation for `µ`. It does turn up in search results, but possibly most of these results are just badly OCRed academic papers. If nobody actually uses this, I can take it out.